### PR TITLE
Build: Java add extra artifacts for individual builds

### PR DESCRIPTION
**Pull request by rozza**
Custom builds for:
* RHEL70
* Ubuntu1604
* osx
* Windows

CDRIVER-3175

### CODE REVIEW

File: bindings/java/mongocrypt/build.gradle.kts [hunk 1]
```diff
@@ -52,7 +52,7 @@ java {
 
 
 dependencies {
-    api("org.mongodb:bson:3.10.1")   // TODO: what version to depend on?
+    api("org.mongodb:bson:[3.10,4.1)")
```
Comment (rozza) [inline bindings/java/mongocrypt/build.gradle.kts]: @jyemin this range should be sufficient enough for now.

File: bindings/java/mongocrypt/build.gradle.kts [hunk 2]
```diff
@@ -62,88 +62,8 @@ dependencies {
 }
 
 /*
- * Publishing
+ * Git version information
  */
-tasks.register<Jar>("sourcesJar") {
```
Comment (rozza) [inline bindings/java/mongocrypt/build.gradle.kts]: These had to be moved until after the classifier tasks were defined.

File: bindings/java/mongocrypt/build.gradle.kts [hunk 3]
```diff
@@ -241,21 +117,30 @@ val copyResources by tasks.register<Copy>("copyResources") {
...
-    "https://s3.amazonaws.com/mciuploads/libmongocrypt/windows-test/master/$revision/libmongocrypt.tar.gz" to "win32-x86-64",
-    "https://s3.amazonaws.com/mciuploads/libmongocrypt/macos/master/$revision/libmongocrypt.tar.gz" to "darwin")
+data class LibMongoCryptS3Data(val evergreenName: String, val classifier: String, val osArch: String) {
+    fun downloadUrl(): String {
+        return "https://s3.amazonaws.com/mciuploads/libmongocrypt/$evergreenName/master/$revision/libmongocrypt.tar.gz"
+    }
+}
+
+// TODO - Any more libraries?
+val jnaMappingList: List<LibMongoCryptS3Data> = listOf(
```
Comment (rozza) [inline bindings/java/mongocrypt/build.gradle.kts]: This is the new mapping

File: bindings/java/mongocrypt/build.gradle.kts [hunk 4]
```diff
@@ -342,3 +243,136 @@ tasks.withType<Test> {
...
+publishing {
+    publications {
+        create<MavenPublication>("mavenJava") {
+            artifactId = "mongocrypt"
+            from(components["java"])
+
+            artifact(tasks["sourcesJar"])
+            artifact(tasks["javadocJar"])
+            // Add special classifier jars
+            tasks.filter { it.name.endsWith("ClassifierJar") }.forEach {
```
Comment (rozza) [inline bindings/java/mongocrypt/build.gradle.kts]: Include the classifier jars

File: bindings/java/mongocrypt/build.gradle.kts (hunk 1 above)
Comment (jyemin) [reply bindings/java/mongocrypt/build.gradle.kts]: We know that 4.0 is compatible, and therefore 4.x will be compatible, so let's make the outer limit 5.0.  That way we don't need to re-publish this when we release the 4.1 driver.

Comment (rozza) [reply bindings/java/mongocrypt/build.gradle.kts]: Done

**Pull request closed without merging**